---
uint8 res
uint8 faultNo